crates/ruff_cli/src/commands/check.rs

```diff
@@ -98,95 +100,113 @@ pub(crate) fn check(
     });
 
     let start = Instant::now();
-    let mut diagnostics: Diagnostics = paths
-        .par_iter()
-        .map(|entry| {
-            match entry {
-                Ok(entry) => {
-                    let path = entry.path();
-                    let package = path
-                        .parent()
-                        .and_then(|parent| package_roots.get(parent))
-                        .and_then(|package| *package);
-
-                    let settings = resolver.resolve(path, pyproject_config);
-
-                    let cache_root = package.unwrap_or_else(|| path.parent().unwrap_or(path));
-                    let cache = caches.as_ref().and_then(|caches| {
-                        if let Some(cache) = caches.get(&cache_root) {
-                            Some(cache)
-                        } else {
-                            debug!("No cache found for {}", cache_root.display());
-                            None
-                        }
-                    });
-
-                    lint_path(
-                        path,
-                        package,
-                        &settings.linter,
-                        cache,
-                        noqa,
-                        fix_mode,
-                        unsafe_fixes,
+    let diagnostics_per_file = paths.par_iter().filter_map(|resolved_file| {
+        let result = match resolved_file {
+            Ok(resolved_file) => {
+                let path = resolved_file.path();
+                let package = path
+                    .parent()
+                    .and_then(|parent| package_roots.get(parent))
+                    .and_then(|package| *package);
+
+                let settings = resolver.resolve(path, pyproject_config);
+
+                if !resolved_file.is_root()
+                    && !&settings.linter.exclude.is_empty()
```

Comment:
nit: the `.is_empty()` check should be inside `match_exclusion`

Reply:
I ended up removing it. `Globset.matches` has an early `is_empty` check.